.eleventy.js

```diff
@@ -147,4 +161,77 @@ module.exports = function (eleventyConfig) {
   eleventyConfig.addFilter("getScriptPath", function (inputPath) {
     return inputPath.split("/").slice(1, -1).join("/") + "/script.js";
   });
+
+  // Rebuild when a change is made to a component template file
+  eleventyConfig.addWatchTarget("src/moj/components/**/*.njk");
+  // Rebuild when gulp has processed component js to the package dir
+  eleventyConfig.addWatchTarget("package/moj/all.js");
+
+   // Allow 11ty to watch and compile the docs site SCSS
+  eleventyConfig.addTemplateFormats("scss");
+  eleventyConfig.addExtension("scss", {
+    outputFileExtension: "css",
+    compile: async function (inputContent, inputPath) {
+      // Skip files like _fileName.scss
+      let parsed = path.parse(inputPath);
+      if (parsed.name.startsWith("_")) {
+        return;
+      }
+
+      // Run file content through Sass
+      let result = sass.compileString(inputContent, {
+        loadPaths: [parsed.dir, ".", "../", "../../", "../../../", "node_modules"],
+        sourceMap: false, // or true, your choice!
+      });
+
+      // Allow included files from @use or @import to
+      // trigger rebuilds when using --incremental
+      this.addDependencies(inputPath, result.loadedUrls);
+
+      return async () => {
+        return result.css;
+      };
+    },
+  });
+
+  // Allow 11ty to watch and parse the docs site JS
+  eleventyConfig.addTemplateFormats('js');
+  eleventyConfig.addExtension('js', {
+    outputFileExtension: 'js',
+    compile: async (content, path) => {
+      // Only process the all.js file, the others are imported
+      if (!path.includes('all')) {
+        return;
+      }
+
+      return async () => {
+        let output = await esbuild.build({
+          target: 'es6',
+          entryPoints: [path],
+          minify: true,
+          bundle: true,
+          write: false,
+        });
+
+        return output.outputFiles[0].text;
+      }
+    }
+  });
+
+  // Copy the docs images to the public assets dir
+  eleventyConfig.addPassthroughCopy( { "docs/assets/images/": "assets/images/"});
```

Comment:
Yeah, I did try this all with `passThroughCopy` but 11ty by default fails the build if two different inputs try to passthrough to the same output. So I moved the copying of the other files to gulp.  

What is the correct order for this? currently gulp runs first, then 11ty so anything in docs assets will overwrite.  Is that correct?

jQuery is handled by gulp too.

Maybe I should just move all the copying into gulp for clarity, I left this in here as I felt like 11ty should be responsible for things related to the docs...

Reply:
I left this within the 11ty config as it will trigger a rebuild if assets are added or modified.  Moving it to gulp would have lost this (or required more setup).